### DISCUSSION TITLE
Pythonnet with Spyder 5.2.2

### DISCUSSION THREAD

**Discussion by abenny10** [pythonnet/pythonnet#2009] [Q&A]
I am having trouble getting pythonnet to run on Spyder 5.2.2 from a .py script. 

When i first open spyder and execute "import clr" in the console I don't get any errors.

However, when I try to execute import clr in a .py script I get the below RuntimeError.

File "C:\Users\username\Anaconda3\lib\site-packages\pythonnet\__init__.py", line 144, in load
    raise RuntimeError("Failed to initialize Python.Runtime.dll")

RuntimeError: Failed to initialize Python.Runtime.dll

Failed to initialize pythonnet: System.InvalidOperationException: This property must be set before runtime is initialized
   at Python.Runtime.Runtime.set_PythonDLL(String value)
   at Python.Runtime.Loader.Initialize(IntPtr data, Int32 size)
   at Python.Runtime.Runtime.set_PythonDLL(String value)
   at Python.Runtime.Loader.Initialize(IntPtr data, Int32 size)

does anyone know why this maybe occurring? I didn't have this issue on a previous version of pythonnet and spyder.

Thanks for any help! 
[A similar question was raised on Stackoverflow ](https://stackoverflow.com/questions/74378613/failed-to-initialize-python-runtime-dll-using-spyder)

**Comment by filmor** [comment]
~~Your installation is broken. The respective code path in the `Loader` is indeed buggy, but it can't be hit from a proper `__init__.py`. How does that file look like for you?~~

Does this also happen when you run the script stand-alone? This can only really happen if the `pythonnet` module gets reloaded and it thus tries to initialize the DLL a second time.

P.S.: You don't have to hide crossposting like this if there is a week between your attempts...

**Reply by abenny10** [reply]
thanks for the response!

Ok I think I understand now! It does not happen if i run standalone. If I run the script first after loading spyder then the import clr does execute. Running the script a second time does create the runtime error. 

How do I check if pythonnet has been previously loaded?

I'm really new to posting on github! What do you mean by hiding crossposting? sorry for the silly question.

**Reply by abenny10** [reply]
from pythonnet import set_runtime
from clr_loader import get_netfx
rt = get_netfx()
set_runtime(rt)

import sys
import clr
